src/clarity-angular/datagrid/datagrid-column-toggle.ts

```diff
@@ -0,0 +1,123 @@
+import { Component, OnDestroy, OnInit } from "@angular/core";
+import { HideableColumnService } from "./providers/hideable-column.service";
+import { DatagridHideableColumn } from "./datagrid-hideable-column";
+import { Subscription } from "rxjs";
+import { Point } from "../popover/popover";
+
+@Component({
+    selector: "clr-dg-column-toggle",
+    template: `
+        <button 
+                #anchor 
+                (click)="toggleUI()"
+                class="btn btn-sm btn-link column-toggle--action">
+            <clr-icon shape="view-columns"></clr-icon>
+        </button>
+        <div class="column-switch"
+             *clrPopover="open; anchor: anchor; anchorPoint: anchorPoint; popoverPoint: popoverPoint">
+            <div class="switch-header">
+                    Show Columns
+                    <button style="min-width: 0; margin: 0; padding: 0"
```

Comment:
Yep - this is actually cleanup I forgot about. Styles are already in the scss file.